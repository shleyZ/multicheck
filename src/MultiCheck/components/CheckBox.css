.CheckboxLabel{
  padding-left: 8px;
  padding-right: 8px;
}

input[type=checkbox]{
  cursor: pointer;
  position: relative;
  width: 15px;
  height: 15px;
  font-size: 14px;
  background-color: #fff;
}

input[type=checkbox]::after{
  position: absolute;
  top: 0;
  background-color: #fff;
  color: #000;
  width: 15px;
  height: 15px;
  display: inline-block;
  visibility: visible;
  padding: 0px;
  text-align: center;
  content: ' ';
  border: 1px solid #999;
}

input[type=checkbox]:checked::after{
  content: "âœ“";
  font-size: 12px;
  font-weight: bold;
  background-color: #000;
  color: #fff;
}