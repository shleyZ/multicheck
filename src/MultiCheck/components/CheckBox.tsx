import React from 'react';

type Props = {
  isChecked?: boolean,
  label?: string,
  value: string,
  handleChanged?: (event: React.ChangeEvent<HTMLInputElement>) => void,
}

const CheckBoxItem: React.FunctionComponent<Props> = (props): JSX.Element => {
  const {handleChanged, isChecked, label} = props;

  return (
    <label>
      <input
        type="checkbox"
        checked={!!isChecked}
        onChange={handleChanged}
        {...props}
      />
      <span>{label}</span>
    </label>
  )
}

export default CheckBoxItem;