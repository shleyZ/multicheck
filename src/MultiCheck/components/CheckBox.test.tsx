
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import 'jsdom-global/register';
import React from 'react';
import CheckBox from './CheckBox';
// TODO more tests

configure({adapter: new Adapter()});

describe('CheckBox', () => {
  describe('initialize', () => {
    it('test the checked props', (done) => {
      const wrapper = mount(
        <CheckBox checked={false} value="no" handleChanged={() => {}} />
      );
      expect(wrapper.find('input[type="checkbox"]').first().props().checked).toBe(false)
      done()
    });

    it('test the value props', (done) => {
      const wrapper = mount(
        <CheckBox checked={false} value="no" handleChanged={() => {}}/>
      );
      expect(wrapper.find('input[type="checkbox"]').first().props().value).toBe('no')
      done()
    });

    it('test the label props', (done) => {
      const wrapper = mount(
        <CheckBox checked={false} value="no" label="status" handleChanged={() => {}}/>
      );
      expect(wrapper.find('span').first().props().children).toBe('status')
      done()
    });
  });

  describe('snapshot', () => {
    it('get snapshot', () => {
      const wrapper = mount(
        <CheckBox checked={false} value="no" handleChanged={() => {}} />
      );
      expect(toJson(wrapper)).toMatchSnapshot();
    })
  })
});
