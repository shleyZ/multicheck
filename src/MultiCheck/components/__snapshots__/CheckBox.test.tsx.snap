// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox snapshot get snapshot 1`] = `
<CheckBoxItem
  checked={false}
  handleChanged={[Function]}
  value="no"
>
  <label
    style={
      Object {
        "display": "block",
      }
    }
  >
    <input
      checked={false}
      onChange={[Function]}
      type="checkbox"
      value="no"
    />
    <span />
  </label>
</CheckBoxItem>
`;
